File: TrackFitApp/src/app/interfaces/workout.ts
export interface Workout {
  workoutId?: number;
  userId?: number;
  exerciseName: string;
  duration: number;
  caloriesBurned?: number;
  workoutDate?: string;
}

export interface PlanExercise {
  name: string;
  sets: number;
  reps: string;
  restTime: string;

  
  bestTime: string;
  difficulty: string;
  whyThisEx: string;
}

export interface WorkoutPlanDay {
  day: string;
  dayName: string;

  
  focus: string;

  isRestDay: boolean;
  exercises: PlanExercise[];
}

File: TrackFitDataAccessLayer/Models/Workout.cs
using System;
using System.Collections.Generic;

namespace TrackFitDataAccessLayer.Models;

public partial class Workout
{
    public int WorkoutId { get; set; }

    public int? UserId { get; set; }

    public string ExerciseName { get; set; }

    public int? Duration { get; set; }

    public int? CaloriesBurned { get; set; }

    public DateTime? WorkoutDate { get; set; }

    public virtual User User { get; set; }
}
